### PR TITLE
Setup pnpm workspace to avoid multiple lockfiles

### LINKED ISSUE #40: Align on either `npm` or `pnpm` for package manager
Both lockfiles exist in this repo and it seems to cause issues with renovate (it's not updating the `pnpm-lock.yaml` file correctly. @ernestoresende did you have a preference here?

### PR DESCRIPTION
**Pull request by ernestoresende**
- closes #40

### CODE REVIEW

**Comment by ernestoresende** [comment]
@trevor-scheer Need help with changing the CircleCI action to look for the correct package manager and lockfile, as it fails in retrieving the lockfile cache. 

Although I moved the entirety of the package root to `/package` workspace, I've put all the scripts to run with the same name trough Turbo, so no further steps should be necessary in order for CI to work (eg.: `test:ci` should run from the root of the project, trough Turbo, so you won't need to change any of the scripts on the action file).

File: .npmrc [hunk 1]
```diff
@@ -1,3 +1,2 @@
 strict-peer-dependencies=true
 auto-install-peers=true
-engine-strict=true
```
Comment (trevor-scheer) [inline .npmrc:3]: Why do we want to remove this?

File: package/src/index.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import { ApolloServer, BaseContext, ContextFunction, HeaderMap } from '@apollo/server';
+import type { ContextFunction } from '@apollo/server';
+import { type ApolloServer, type BaseContext, HeaderMap } from '@apollo/server';
```
Comment (trevor-scheer) [inline package/src/index.ts:2]: Can be combined

**Comment by trevor-scheer** [comment]
@ernestoresende got CI passing with pnpm, so just a couple comments to address 👍

**Comment by ernestoresende** [comment]
@trevor-scheer All good, next from here will be adding `changesets` and configuring canary release, and we'll be golden for v1 ✨

**Review by trevor-scheer — APPROVED** [review]

**Pull request merged**